ChromecastDemo/app/src/main/java/com/jwplayer/chromecastdemo/JWPlayerViewExample.java

```diff
@@ -0,0 +1,159 @@
+package com.jwplayer.chromecastdemo;
+
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.content.res.Configuration;
+import android.os.Bundle;
+import android.view.KeyEvent;
+import android.view.Menu;
+
+import com.google.android.gms.cast.framework.CastButtonFactory;
+import com.google.android.gms.cast.framework.CastContext;
+import com.google.android.gms.cast.framework.CastState;
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.GoogleApiAvailability;
+import com.longtailvideo.jwplayer.JWPlayerView;
+import com.longtailvideo.jwplayer.events.FullscreenEvent;
+import com.longtailvideo.jwplayer.events.listeners.VideoPlayerEvents;
+import com.longtailvideo.jwplayer.media.playlists.PlaylistItem;
+
+import java.util.List;
+
+import androidx.appcompat.app.ActionBar;
+import androidx.appcompat.app.AppCompatActivity;
+
+
+public class JWPlayerViewExample extends AppCompatActivity
+		implements VideoPlayerEvents.OnFullscreenListener {
+
+	private JWPlayerView mPlayerView;
+
+	private CastContext mCastContext;
+
+	private static final String GooglePlayStorePackageNameOld = "com.google.market";
+	private static final String GooglePlayStorePackageNameNew = "com.android.vending";
+
+	@Override
+	protected void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.activity_jwplayerview);
+
+		if (isGoogleApiAvailable(this)) {
+			mCastContext = CastContext.getSharedInstance(getApplicationContext());
+		}
+
+		mPlayerView = findViewById(R.id.jwplayer);
+
+		// Handle hiding/showing of ActionBar
+		mPlayerView.addOnFullscreenListener(this);
+
+		// Keep the screen on during playback
+		new KeepScreenOnHandler(mPlayerView, getWindow());
+
+		// Load a media source
+		PlaylistItem pi = new PlaylistItem.Builder()
+				.file("https://content.jwplatform.com/manifests/1sc0kL2N.m3u8")
+				.title("Press Play")
+				.image("https://content.jwplatform.com/thumbs/1sc0kL2N.jpg")
+				.description("Press play with JW Player")
+				.build();
+
+		mPlayerView.load(pi);
+
+
+	}
+
+	// Without the Google API's Chromecast won't work
+	private boolean isGoogleApiAvailable(Context context) {
```

Comment:
Do we really need to check for the playstore being installed? I think GoogleApiAvailability takes care of that. Also the flag GET_UNINSTALLED_PACKAGES has been deprecated, check: https://developer.android.com/reference/android/content/pm/PackageManager.html#GET_UNINSTALLED_PACKAGES

Reply:
@egonzalez-jw sounds good, I was following the same pattern we use in our demo app. Google APIs need to be installed for Chromecast to work but I agree we can use a better method for this. I'll patch it up

Reply:
@egonzalez-jw I changed the logic to only look for the specific packages rather than get all of them. Technically, the player already knows this info, it would be nice to expose that via our API since we already check. But for now this should be good